### PR TITLE
SCMOD-5975: Addressing review comments

### PR DESCRIPTION
**Pull request by kusumaghoshdastidar**
1. Collapsed MultiPartDocument and MultiPartDocumentResource together in the same way and removed "String name" from the parameters.
2. Collapsed MultiPartFile and MultiPartContentResource.
3. Changed createOrReplaceBatch to throw ApiException only
4. Removed "FileNotFoundException" throw from the MultiPart interface
5. Set the User-Agent, couldn't set it using the code generator
6. Changed the response for the PUT request

### CODE REVIEW

File: staging-service-client/pom.xml [hunk 1]
```diff
@@ -89,7 +93,7 @@
                             <apiPackage>com.github.cafdataprocessing.services.staging.client.internal</apiPackage>
                             <invokerPackage>com.github.cafdataprocessing.services.staging.client</invokerPackage>
                             <modelNamePrefix>Staging</modelNamePrefix>
-                            <!--<ignoreFileOverride>${project.basedir}/src/main/swagger-config/.swagger-codegen-ignore</ignoreFileOverride>-->
+                            <httpUserAgent>caf-staging-service-java-client/${project.version}</httpUserAgent><!-- does not work-->
```
Comment (dermot-hardy) [inline staging-service-client/pom.xml]: This says does not work.

File: staging-service-client/src/main/java/com/github/cafdataprocessing/services/staging/client/StagingApi.java [hunk 2]
```diff
@@ -38,10 +35,34 @@
...
+                        @Override
+                        public Response intercept(final Chain chain) throws IOException {
+                            final Request request = chain.request();
+                            final Request newRequest = request.newBuilder()
+                                    .header(USER_AGENT_HEADER_NAME, USER_AGENT_HEADER_VALUE)
+                                    .build();
+                            return chain.proceed(newRequest);
+                        }
+                    }
+                );*/
```
Comment (dermot-hardy) [inline staging-service-client/src/main/java/com/github/cafdataprocessing/services/staging/client/StagingApi.java]: Why is this code commented out?  Remove it if it is not used.

File: staging-service-client/src/main/java/com/github/cafdataprocessing/services/staging/client/StagingApi.java [hunk 3]
```diff
@@ -38,10 +35,34 @@
  *
  */
 public class StagingApi extends com.github.cafdataprocessing.services.staging.client.internal.StagingApi {
+    private final String USER_AGENT_HEADER_NAME = "User-Agent";
+    private final String USER_AGENT_HEADER_VALUE = "caf-staging-service-java-client/1.0.0-SNAPSHOT";
```
Comment (dermot-hardy) [inline staging-service-client/src/main/java/com/github/cafdataprocessing/services/staging/client/StagingApi.java]: Is the version number hard-coded?

**Review by dermot-hardy — CHANGES_REQUESTED** [review]

File: staging-service-client/pom.xml (hunk 1 above)
Comment (kusumaghoshdastidar) [reply staging-service-client/pom.xml]: Removed

File: staging-service-client/src/main/java/com/github/cafdataprocessing/services/staging/client/StagingApi.java (hunk 3 above)
Comment (kusumaghoshdastidar) [reply staging-service-client/src/main/java/com/github/cafdataprocessing/services/staging/client/StagingApi.java]: Removed. Using the default user-agent set by code generator instead

File: staging-service-client/src/main/java/com/github/cafdataprocessing/services/staging/client/StagingApi.java (hunk 2 above)
Comment (kusumaghoshdastidar) [reply staging-service-client/src/main/java/com/github/cafdataprocessing/services/staging/client/StagingApi.java]: Removed.

**Review by dermot-hardy — APPROVED** [review]

**Pull request merged**